### PR TITLE
Fixed issue with missing access token and no UserInfo call

### PR DESCRIPTION
**Pull request by xperiandri**
Fixed UserInfo endpoint not being called even if GetClaimsFromUserInfoEndpoint is set to true

### CODE REVIEW

**Comment by jennyf19** [comment]
We need MSAL to acquire the token and keep it in its cache, so that the token can be refreshed by MSAL. This is by design.

**Comment by xperiandri** [comment]
Then fix the ability to call UserInfo endpoint the way you designed

**Comment by xperiandri** [comment]
Along with that the second option does not work too
See https://github.com/AzureAD/microsoft-identity-web/discussions/1773

**Review by jmprieur — COMMENTED** [review]
I don't think that this is the right fix

File: src/Microsoft.Identity.Web/TokenAcquisition.cs [hunk 1]
```diff
@@ -226,7 +226,7 @@ internal MergedOptions GetOptions(string? authenticationScheme, out string effec
                 var result = await builder.ExecuteAsync()
                                           .ConfigureAwait(false);
 
-                context.HandleCodeRedemption(null, result.IdToken);
+                context.HandleCodeRedemption(result.AccessToken, result.IdToken);
```
[the remark this answers is not in the document]
Comment (xperiandri) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: But the default ASP.NET Core logic expects the AccessToken in order to call UserInfo endpoint if `GetClaimsFromUserInfoEndpoint=true`. There are only 2 checks:
1. If `GetClaimsFromUserInfoEndpoint=true`
2. If access token avaialble
As long as you remove access token, the second check never succeeds and UserInfo endpoint will never be called

[the remark this answers is not in the document]
Comment (xperiandri) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: But it has pretty useful claims, like first and last name
``` JSON
{
  "sub": "0_MF9yUOWfJpPXtwidxYGuQB7qPi1-CVWCPJOZGXsok",
  "name": "Андрій Чебукін",
  "family_name": "Чебукін",
  "given_name": "Андрій",
  "picture": "https://graph.microsoft.com/v1.0/me/photo/$value",
  "email": "xperiandri@secretcircle.solutions"
}
```

[the remark this answers is not in the document]
Comment (xperiandri) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: I don't mind if it is not refreshable. I only want that UserInfo endpoint being called no matter how 🙂

[the remark this answers is not in the document]
Comment (jmprieur) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: Oh that's not what we get from the user info endpoint (which only gets you the home tenant Id and home object ID of the signed-in user).

What you need is the ID token and user profile, that you get with graph (me endpoint)

[the remark this answers is not in the document]
Comment (jmprieur) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: Also note that, in a web App, you should **NEVER** look at the content of the access token. The access token is for the web API that will be called with this token. It's **NOT** for the client (and could be encrypted)

[the remark this answers is not in the document]
Comment (xperiandri) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: That is a real response. When I debugged your code, I put the access token back, where you throw it away. And I get that response

[the remark this answers is not in the document]
Comment (jmprieur) [reply src/Microsoft.Identity.Web/TokenAcquisition.cs:229]: @jennyf19 
We should test this change with:
- AAD apps with and without downstream APIs
- B2c apps with and without downstream APIs

**Comment by jmprieur** [comment]
@xperiandri, closing this PR, as we've provided the fix in Microsoft.Identity.Web 2.1.0
The fix is in master if you want to try it out.

**Comment by xperiandri** [comment]
Great, thanks!

**Pull request closed without merging**